### ISSUE TITLE
layout is masked from ‘package:graphics’

### ISSUE THREAD

**Issue by talgalili** [ropensci/plotly#464]
I just noticed that there is a plotly::layout function. This is a recipe for clashes with other packages which use graphics (in their DESCRIPTION) as depends (instead of as import).
I just had this clash between plotly and gplots.

I think the function should be renamed (maybe layout_ly or something like that), or be hidden from the global environment.

p.s.: I am surprised this didn't come up on CRAN. Packages should avoid masking functions from the base R packages as much as possible.

With regards,
Tal

**Comment by talgalili** [comment]
@cpsievert 
I see you're fixing many things on plotly. Any chance to give this a look?

**Comment by cpsievert** [comment]
I agree, it was a mistake, but at this point, I think the breaking changes outweigh the benefits

**Comment by talgalili** [comment]
I think it could be solved (probably without breaking anything) by turning it into S3. For example:

# first you'll need to move layout to some other name
layout_ly <- layout

And then use:


```
layout <- function (p, ...) {
     UseMethod("layout")
}

layout.default <- function(p, ...) {
# you may wish to add some code here in case someone wishes to use mat explicitly and not mention p in any way. So this would need to be fixed - the question is if you think this direction would solve the isse.
     graphics::layout(mat=p, ...)
}

layout.plotly<- function(p, ...) {
     plotly::layout_ly(p=p, ...)
}

```

You may prefer to use other names for the first parameter (since in

**Comment by cpsievert** [comment]
Hmm, interesting, I'll reopen and give it some more thought

**Comment by talgalili** [comment]
Thanks.

I used a similar trick in dendextend for dealing with cutree (I wanted to have a cutree.dendrogram and created a cutree.hclust from stats::cutree). See line 691:
https://github.com/talgalili/dendextend/blob/master/R/cutree.dendrogram.R

**Comment by timelyportfolio** [comment]
I think this would be a good approach.  @cpsievert, let me know if you want me to explore it.

**Comment by cpsievert** [comment]
Closed via d2bf2a1eb56d824a091e2cc9afea09a5968dd050. Let us know if you're still having problems, thanks!

**Comment by talgalili** [comment]
Great, thanks :)
